src/core/load-fragment.c

```diff
@@ -3939,7 +3941,14 @@ int config_parse_bind_paths(
                         return 0;
                 }
 
-                s = source;
+                r = unit_full_printf(u, source, &sresolved);
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r,
+                                   "Failed to resolved specifiers in \"%s\", ignoring: %m", source);
+                        return 0;
```

Comment:
so, strictly speaking, @evverx is right: we should catch ENOMEM here explicitly, and then call log_oom() for it, and return an error, as ENOMEM is not a user issue, but a non-recoverable error. That said, we have been sloppy with this in general, so I figure it's OK if this patch doesn't do this either